node/silkworm/db/storage.hpp

```diff
@@ -15,88 +15,92 @@
-            }
-        }
-
-        return short_form + long_form;
+class BlockAmount {
+  public:
+    enum Type {
+        kOlder,  // Prune Data Older than (moving window)
+        kBefore  // Prune data before (fixed)
+    };
+    explicit BlockAmount() : value_{std::nullopt}, enabled_{false}, type_{Type::kOlder} {};
+    explicit BlockAmount(Type type, BlockNum value) : value_{value}, enabled_{true}, type_{type} {};
+
+    [[nodiscard]] bool enabled() const { return enabled_; }
+    [[nodiscard]] Type type() const { return type_; };
+    [[nodiscard]] BlockNum value() const;
+    [[nodiscard]] BlockNum value_from_head(BlockNum stage_head) const;
+
+    void to_string(std::string& short_form, std::string& long_form, char prefix) const;
+
+    bool operator==(const BlockAmount& other) const {
+        return enabled_ == other.enabled_ && type_ == other.type_ && value_ == other.value_;
     }
+    bool operator!=(const BlockAmount& other) const { return !(this->operator==(other)); }
+
+  private:
+    const std::optional<BlockNum> value_;
+    const bool enabled_;
+    const Type type_;
+};
+
+class PruneMode {
+  public:
+    explicit PruneMode() : initialized_{false}, history_(), receipts_(), tx_index_(), call_traces_(){};
+    explicit PruneMode(bool initialized)
+        : initialized_{initialized}, history_(), receipts_(), tx_index_(), call_traces_(){};
+    explicit PruneMode(BlockAmount& history, BlockAmount& receipts, BlockAmount& tx_index, BlockAmount& call_traces)
+        : initialized_{true},
+          history_{std::move(history)},
+          receipts_{std::move(receipts)},
+          tx_index_{std::move(tx_index)},
+          call_traces_{std::move(call_traces)} {};
+    [[nodiscard]] bool initialized() const { return initialized_; }
+    [[nodiscard]] const BlockAmount& history() const { return history_; }
+    [[nodiscard]] const BlockAmount& receipts() const { return receipts_; }
+    [[nodiscard]] const BlockAmount& tx_index() const { return tx_index_; }
+    [[nodiscard]] const BlockAmount& call_traces() const { return call_traces_; }
+
+    [[nodiscard]] std::string to_string() const;
 
     bool operator==(const PruneMode& other) const {
-        return history == other.history && receipts == other.receipts && tx_index == other.tx_index &&
-               call_traces == other.call_traces;
+        return history_ == other.history_ && receipts_ == other.receipts_ && tx_index_ == other.tx_index_ &&
+               call_traces_ == other.call_traces_;
     }
     bool operator!=(const PruneMode& other) const { return !(this->operator==(other)); }
-};
 
-inline constexpr PruneMode kDefaultPruneMode{
-    true,          // initialized
-    std::nullopt,  // history
-    std::nullopt,  // receipts
-    std::nullopt,  // tx_index
-    std::nullopt   // call_traces
+  private:
+    const bool initialized_;         // Whether this class is initialized or default
```

Comment:
I don't like initialized for C++ objects. IMHO if the constructor succeeds, the object is initialised. If the constructor doesn't succeed (i.e. the constructor throws because, for example, a class invariant is not satisfied), no object is constructed. I don't see a point in having a special in-between state "constructed-but-uninitialized".